Задание 8

Temkame opened the pull request.

Temkame committed:
Задание 8

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+     * @return Дневной лимит
+     */
+    @GetMapping("/{userId}")
+    public ResponseEntity<BigDecimal> getDailyLimit(@PathVariable Long userId) {
```
весь курс я оставляю комментарии по поводу ResponseEntity

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+     * @return Ответ с сообщением об успехе
+     */
+    @PostMapping("/{userId}/reduce")
+    public ResponseEntity<String> reduceLimit(
```
ResponseEntity< String > - строка не может быть объектом контракта

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+     * @return Ответ с сообщением об успехе
+     */
+    @PostMapping("/{userId}/restore")
+    public ResponseEntity<String> restoreLimit(
```
см выше

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+        try {
+            userLimitService.reduceLimit(userId, amount);
+            return ResponseEntity.ok("Limit reduced successfully for user with ID: " + userId);
+        } catch (LimitExceededException e) {
```
обработка ошибок должна быть в RestControllerAdvice, а выброс их в userLimitService

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+            userLimitService.reduceLimit(userId, amount);
+            return ResponseEntity.ok("Limit reduced successfully for user with ID: " + userId);
+        } catch (LimitExceededException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
```
это не логика контроллера

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+            userLimitService.restoreLimit(userId, amount);
+            return ResponseEntity.ok("Limit restored successfully for user with ID: " + userId);
+        } catch (UserNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
```
см выше

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/controller/UserLimitController.java:
```diff
@@ -0,0 +1,85 @@
+    @PostMapping("/reset")
+    public ResponseEntity<String> resetAllLimits() {
+        userLimitService.resetAllLimits();
+        return ResponseEntity.ok("All limits reset to default value: 10000.00");
```
строка не может быть контрактом

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/service/UserLimitService.java:
```diff
@@ -0,0 +1,92 @@
+@Service
+public class UserLimitService {
+
+    private static final BigDecimal DEFAULT_LIMIT = new BigDecimal("10000.00");
```
статики быть не должно! и это параметр нужно вынести в properties

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/service/UserLimitService.java:
```diff
@@ -0,0 +1,92 @@
+     */
+    @Transactional
+    public void resetAllLimits() {
+        userLimitRepository.findAll().forEach(userLimit -> {
```
множестенные запросы в бд, не оптимально

Alexandr056 commented on src/main/java/ru/fokin/java_pro_limitservice/service/UserLimitService.java:
```diff
@@ -0,0 +1,92 @@
+import java.math.BigDecimal;
+import java.util.Optional;
+
+@Service
```
нет логов

Alexandr056 reviewed (changes requested):
нужны доработки

Temkame committed:
1. Обработку ошибок сделал в RestControllerAdvice, а выброс их в userLimitService.
2. Поменял логику для контроллера, убрал возвращение строки в ResponseEntity, сделал возврат объекта.
3. Добавил DTO LimitResponse
4. Убрал статику быть из сервиса и это параметр вынес в properties
5. Добавил логи в сервсис
